### DISCUSSION TITLE
how to hide resolvers(like application schema)

### DISCUSSION THREAD

**Discussion by omar-dulaimi** [MichalLytek/typegraphql-prisma#71] [Q&A] [answered]
Hello,

Coming from the Prisma 1 world to TypeGraphQL :) 
I was just wondering if my use case is doable through typegraphql and prisma 2.

Back in Prisma 1, we had two graphql schemas; one of them was generated and had tons of types, etc. The other one was maintained by the user and was called Application schema.

Now we used to use the application schema to hide some resolvers from the outside world and just expose a few.

Since we currently use the typegraphql prisma genertaor,  it generates all possible resolvers, many of them we would like to hide.

Is that use case possible? 

And since our previous resolvers had custom logic before saving/deleting/etc, is it possible to have that logic back?

Thanks a lot!

**Comment by MichalLytek** [comment, marked as answer]
> Now we used to use the application schema to hide some resolvers from the outside world and just expose a few.
> Since we currently use the typegraphql prisma genertaor, it generates all possible resolvers, many of them we would like to hide.
> Is that use case possible?

You can pick the actions you want:
https://github.com/MichalLytek/typegraphql-prisma#exposing-selected-prisma-actions-only

Or have two separate schemas (private and public one):

```ts
const firstSchema = await buildSchema({
  resolvers: [UserCrudResolver],
});
const secondSchema = await buildSchema({
  resolvers: [CategoryUpdateResolver],
});
```      

> And since our previous resolvers had custom logic before saving/deleting/etc, is it possible to have that logic back?

TypeGraphQL hasn't provided any integration with Prisma 1, so I'm not sure about what support should I bring back and what do you expect from this `typegraphql-prisma` generator.

**Reply by omar-dulaimi** [reply]
Could you elaborate on using private and public schemas
what would the next step be after separating them

**Reply by MichalLytek** [reply]
exposing on two different ports via Apollo Server

**Reply by MichalLytek** [reply]
or under one port using custom http server with routing based on header or something

**Reply by omar-dulaimi** [reply]
Would that work fine(or is considered a best practice) when using a serverless env like lambda? running two schemas on two different ports?

**Reply by MichalLytek** [reply]
they would be just a two separate apps with two separate entry points, just with shared part of the codebase

**Reply by omar-dulaimi** [reply]
thanks
